cli_main.py

```diff
@@ -69,6 +69,47 @@ def _warning(message,
 warnings.showwarning = _warning
 
 
+def convert_str_time(x, dt):
+    """
+    Converts a string x into a frame number based on given `dt`.
+    If `x` does not contain any units its assumed to be a frame number already.
+
+    Parameters
+    ----------
+    x : str
+        the input string
+    dt : float
+        the time step in ps
+
+    Returns
+    -------
+    int
+        frame number
+
+    Raises
+    ------
+    ValueError
+        The input does not contain any units but is not an integer.
+    """
+
+    # regex to split value and units while handling scientific input
+    type_regex = re.compile(r"(?!e)[a-z]")
```

Comment:
Please correct me if I am missing some detail, what do you thing on using

```python
re.compile(r"(?!e)\d+|\D+")
```

It has the following behaviour, and avoids the cryptic part in line 100 and 101:

https://regex101.com/r/XLT58l/1

Then it could go something like:

```python
try:
    if len(res) == 2:
        val, unit = res
        # the rest is just as you did
    else:
        frame = int(x)
except (TypeError, ValueError):
    raise ValueError("Only integers or time step combinations (´12ps´) are valid for frame selection")
```

Reply:
This is really nice! However, we have to adjust this a bit to parse inputs like `1e5ps` or `1E5ps`. You are really got in writing regular expressions. Do you have a idea?

Reply:
Hi @joaomcteixeira,

thanks for the good comments. The regex is really nice. We just have to improve it a bit 🙃.

There is no rush here. Take your time and finish your important projects!

Reply:
Jajaja, :stuck_out_tongue_closed_eyes: 
I will consider myself a programmer once I've mastered regexes... :rofl: 

This is the best I could find in this half an hour, what do you think? Does it match all the cases? It is not the prettiest I would like to have, I admit. Maybe now I am just over complicating. :roll_eyes: To be continued...

```python
strs = [
    '1',
    '-1',
    '12e3',
    '12e-14',
    '12e5ps'
]

r = re.compile(r'((-?\d{1,}e?-?\d*)|[a-z]*$)')

for s in strs:
    val_, unit_, *_ = r.findall(s)
    val, unit = val_[0], unit_[0]
    print(val, f'{unit!r}')
```

```
1 ''
-1 ''
12e3 ''
12e-14 ''
12e5 ps
```

Reply:
Not the prettiest but it is working smoothly. I updated the PR.